src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java

```diff
@@ -0,0 +1,133 @@
+package de.zalando.aruha.nakadi.controller;
+
+import de.zalando.aruha.nakadi.NakadiException;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.problem.DuplicatedEventTypeNameProblem;
+import de.zalando.aruha.nakadi.problem.ValidationProblem;
+import de.zalando.aruha.nakadi.repository.DuplicatedEventTypeNameException;
+import de.zalando.aruha.nakadi.repository.EventTypeRepository;
+import de.zalando.aruha.nakadi.repository.NoSuchEventTypeException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.*;
+import org.zalando.problem.MoreStatus;
+import org.zalando.problem.Problem;
+
+import javax.validation.Valid;
+import javax.ws.rs.core.Response;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.springframework.http.ResponseEntity.status;
+
+@RestController
+@RequestMapping(value = "/event_types")
+public class EventTypeController {
```

Comment:
I think there is too much logic in private methods in this class. Maybe extracting it to a service class would be better?